internal/ls/findallreferences.go

```diff
@@ -410,22 +406,53 @@ func (l *LanguageService) ProvideReferences(params *lsproto.ReferenceParams) []*
 
 	symbolsAndEntries := l.getReferencedSymbolsForNode(position, node, program, program.GetSourceFiles(), options, nil)
 
-	return core.FlatMap(symbolsAndEntries, l.convertSymbolAndEntryToLocation)
+	return core.FlatMap(symbolsAndEntries, l.convertSymbolAndEntriesToLocations)
+}
+
+func (l *LanguageService) ProvideImplementations(params *lsproto.ImplementationParams) []*lsproto.Location {
+	program, sourceFile := l.getProgramAndFile(params.TextDocument.Uri)
+	position := int(l.converters.LineAndCharacterToPosition(sourceFile, params.Position))
+	node := astnav.GetTouchingPropertyName(sourceFile, position)
+
+	var seenNodes collections.Set[*ast.Node]
+	var entries []*referenceEntry
+	queue := l.getImplementationReferenceEntries(program, node, position)
```

Comment:
The original had something like the following check before processing as a queue

```
    if (
        node.parent.kind === SyntaxKind.PropertyAccessExpression
        || node.parent.kind === SyntaxKind.BindingElement
        || node.parent.kind === SyntaxKind.ElementAccessExpression
        || node.kind === SyntaxKind.SuperKeyword
    ) {
        referenceEntries = entries && [...entries];
    }
```

and idea what that was intended for? I think there was some special handling for not "cascading" beyond the direct superclass

Reply:
Yeah, I think it is so that Go To Implementation on say a `super` keyword only goes to that type and not other types that implement it. Which seems a little odd. Not really sure why you'd want to Go To Implementation on `super`, as opposed to Go To Definition or Go To Type Definition.

Reply:
I feel like that makes sense, though, since super is basically "do the parent class thing" and so shouldn't jump anywhere else?

Reply:
My mental model of it is "the implementation of `super` is code in a parent class's constructor".

Reply:
Yeah I looked at the test cases and was surprised. Both go-to-definition and go-to-implementation should probably go to the constructor, but instead we go to the base class for go-to-implementation which feels like it is the less-intuitive thing.